src/kaltura-player.js

```diff
@@ -36,19 +39,38 @@ class KalturaPlayer extends FakeEventTarget {
   _provider: Provider;
   _uiWrapper: UIWrapper;
   _logger: any;
+  _pluginManager: PluginManager;
+  _controllerProvider: ControllerProvider;
+  _adsController: ?AdsController;
+  _pluginsConfig: KPPluginsConfigObject;
+  _pluginEvents: {[plugin: string]: {[event: string]: string}};
+  _pluginsUiComponents: Array<KPUIComponent>;
+  _reset: boolean;
+  _sourceSelected: boolean;
 
   constructor(options: KPOptionsObject) {
     super();
     this._eventManager = new EventManager();
-    const {sources} = options;
+    const {sources, plugins} = options;
     const noSourcesOptions = Utils.Object.mergeDeep({}, options, {sources: null});
+    delete noSourcesOptions.plugins;
     this._localPlayer = loadPlayer(noSourcesOptions);
     this._logger = getLogger('KalturaPlayer' + Utils.Generator.uniqueId(5));
+    this._reset = true;
+    this._pluginsConfig = {};
+    this._pluginEvents = {};
+    this._pluginManager = new PluginManager();
+    this._controllerProvider = new ControllerProvider(this._pluginManager);
+    this._pluginsUiComponents = [];
+    this.configure({plugins});
     this._uiWrapper = new UIWrapper(this, options);
     this._provider = new Provider(options.provider, __VERSION__);
     this._playlistManager = new PlaylistManager(this, options);
     this._playlistManager.configure(options.playlist);
     Object.values(CoreEventType).forEach(coreEvent => this._eventManager.listen(this._localPlayer, coreEvent, e => this.dispatchEvent(e)));
+    this._eventManager.listen(this, CoreEventType.CHANGE_SOURCE_STARTED, () => this._onChangeSourceStarted());
+    this._eventManager.listen(this, CoreEventType.ENDED, () => this._onEnded());
+    this._eventManager.listen(this, CoreEventType.SOURCE_SELECTED, () => (this._sourceSelected = true));
```

Comment:
won't this be the same as setting it when setMedia is called or when sources are passed via constructor?
and actually, won't it be safer and more correct to pass via setMedia if sources are passed via constructor?

Reply:
for 1. `SOURCE_SELECTED` is fired only when there is actually source to play (hls, dash or progressive). i don't want to add this validation also here.
for 2. how it's related to this change